src/wallet/api/wallet_manager.cpp

```diff
@@ -114,6 +114,27 @@ Wallet *WalletManagerImpl::createWalletFromKeys(const std::string &path,
     return wallet;
 }
 
+Wallet *WalletManagerImpl::createWalletFromDevice(const std::string &path,
+                                                  const std::string &password,
+                                                  NetworkType nettype,
+                                                  const std::string &deviceName,
+                                                  uint64_t restoreHeight,
+                                                  const std::string &subaddressLookahead)
+{
+    WalletImpl * wallet = new WalletImpl(nettype);
+    if(restoreHeight > 0){
+        wallet->setRefreshFromBlockHeight(restoreHeight);
+    }
+    auto lookahead = tools::parse_subaddress_lookahead(subaddressLookahead);
+    if (!lookahead)
+    {
+        lookahead = tools::parse_subaddress_lookahead("5:20");
+    }
+    wallet->setSubaddressLookahead(lookahead->first, lookahead->second);
```

Comment:
Though with this, i'll override the wallet2 settings even when not specified. The defaults should be in wallet2 so other programs benefit from a good value.

Reply:
OK, please see the first commit.